plainly-plugin/src/ui/components/form/UploadForm.tsx

```diff
@@ -29,40 +30,44 @@ export default function UploadForm() {
   const { isPending: isEditing, mutateAsync: editProject } = useEditProject();
   const { notifySuccess, notifyError } = useNotifications();
 
-  const projectExists = !!(projectData?.id && data);
-  const removedFromDatabase = !!(projectData?.id && !data);
-
-  const revisionHistoryCount = data?.revisionHistory?.length || 0;
-  const badRevision =
-    projectExists && projectData?.revisionCount !== revisionHistoryCount;
-
   const [inputs, setInputs] = useState<{
     projectName?: string;
     description?: string;
     tags?: string[];
   }>({});
   const [uploadMode, setUploadMode] = useState<'new' | 'edit'>();
 
+  const localProjectExists = !!projectData?.id;
+  const remoteProjectExists = !!(localProjectExists && data);
+
   const uploadModes = [
     {
       value: 'new',
       label: 'Upload new',
-      checked: uploadMode === 'new' || !projectExists,
+      checked: uploadMode === 'new' || !remoteProjectExists,
```

Comment:
I don't think this is good can you actually swicth to new now once you have existing project?  I think the or condition must be used only if upload mode is not defined  `(uploadMode === undefined && ..)`